### DISCUSSION TITLE
Pagination support

### DISCUSSION THREAD

**Discussion by kathir-arizon** [bigcommerce/catalyst#1689] [General]
Hi Team,

How do we display the pagination with total records on PLP & SLP?

Based on Graphql, It is returning the cursor value, and handling the pagination without total records will be challenging. We are not using Algolia in one of our projects. Kindly suggest.

**Comment by chanceaclark** [comment]
Unfortunately it's not really possible with cursor based pagination. We'd have to introduce offset based pagination in our GraphQL to support this. However, that's a massive effort and has performance impacts. We've been chatting about this internally, but nothing to make a decision and/or start that effort anytime soon.

**Reply by kathir-arizon** [reply]
Thanks @chanceaclark  for the update.
